flytekit/experimental/eager_function.py

```diff
@@ -605,16 +617,28 @@ def _internal_demo_remote(remote: FlyteRemote) -> FlyteRemote:
 
 def _internal_remote(
     remote: FlyteRemote,
-    client_secret_group: str,
-    client_secret_key: str,
+    client_secret_group: Optional[str],
+    client_secret_key: Optional[str],
+    client_secret_env_var: Optional[str],
 ) -> FlyteRemote:
     """Derives a FlyteRemote object from a yaml configuration file, modifying parts to make it work internally."""
-    assert client_secret_group is not None, "secret_group must be defined when using a remote cluster"
-    assert client_secret_key is not None, "secret_key must be defined a remote cluster"
     secrets_manager = current_context().secrets
     client_secret = secrets_manager.get(client_secret_group, client_secret_key)
     # get the raw output prefix from the context that's set from the pyflyte-execute entrypoint
     # (see flytekit/bin/entrypoint.py)
+
+    if client_secret_env_var is not None:
+        # this creates a remote client where the env var client secret is sufficient for authentication
+        os.environ[client_secret_env_var] = client_secret
```

Comment:
confused what's happening here.  if we're removing the asserts, then that means the group/key might be None.  If they're none, won't the secrets_manager.get fail?  So then what is the client_secret?  Or is there some default in the secrets manager that somehow gets returned?
I don't get why we're setting the environment variable here to the value of client secret.

Reply:
this is to make it easy to authenticate with `UnionRemote` on serverless (via `UNION_SERVERLESS_API_KEY ` env var). Secrets on serverless only needs a secret key (no group).

Reply:
just wondering, in the following scenario:
* `client_secret_key` is  `None`
* `client_secret_group` is `None`
(and this scenario is now valid because the asserts are now removed)
after line 626 `client_secret = secrets_manager.get(client_secret_group, client_secret_key)`
what is the value of `client_secret`?

Reply:
going through the secretsmanager, it looks like it raises an IndexError if you call `get(group=None, key=None)`, so I don't think it ever gets past line 626.  Could we move the assertion that at least one of those has be be non-None into the `get` function?